doc: reload pods when using the same DAG tag

sybernatus opened the pull request:
Using the same tag for airflow dags cannot work as kubernetes will not refresh pods (scheduler, workers, webservers) if they are not updated.
Then you need to trigger the reloading of those pods manually.

<!--
Thank you for contributing! Please make sure that your code changes
are covered with tests. And in case of new features or big changes
remember to adjust the documentation.

Feel free to ping committers for the review!

In case of existing issue, reference it using one of the following:

closes: #ISSUE
related: #ISSUE

How to write a good git commit message:
http://chris.beams.io/posts/git-commit/
-->

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragement file, named `{pr_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

sybernatus committed:
doc: reload pods when using the same DAG tag

Using the same tag for airflow dags cannot work as kubernetes will not refresh pods (scheduler, workers, webservers) if they are not updated.
Then you need to trigger the reloading of those pods manually.

potiuk commented on docs/helm-chart/manage-dags-files.rst:
```diff
@@ -77,6 +77,12 @@ If you are deploying an image with a constant tag, you need to make sure that th
       --set images.airflow.repository=my-company/airflow \
       --set images.airflow.tag=8a0da78 \
       --set images.airflow.pullPolicy=Always
+      
```
I would add a message discouraging use of constant tag and add this description below as an example of problems you might get when you do.

Using constant image tags and re-uploading them is an extremely bad practice for production. you are not supposed to do it because you are completely loosing any traceability of what happened. It's equivalent of randomly changing your code in production server without keeping it in any kind of source repository history. Basically i cannot imagine any scenario where it would be acceptable other than one-man-run-airflow for own personal usage for testing/development.

sybernatus committed:
Add warning for constant tag usage

uranusjr commented on docs/helm-chart/manage-dags-files.rst:
```diff
@@ -71,12 +71,22 @@ Finally, update the Airflow pods with that image:
 
 If you are deploying an image with a constant tag, you need to make sure that the image is pulled every time.
 
+Warning :warning:
```
There is a syntax for this:

```rst
.. warning::
```

jedcunningham commented on docs/helm-chart/manage-dags-files.rst:
```diff
@@ -71,12 +71,22 @@ Finally, update the Airflow pods with that image:
+
+.. code-block:: bash
+
+  "kubectl" delete pod "<airflow-webserver>" "<airflow-scheduler>" "<airflow-worker>"
```
```suggestion
      --set images.airflow.pullPolicy=Always
      --set airflowPodAnnotations.random=r${RANDOM}
```

You'd be much better off doing this instead, or using `openssl rand -hex 8`, or something (you get the idea).

This will recreate all of the pods without requiring manual intervention, but for sure heed @potiuk's warning.

sybernatus committed:
doc: fix rst warning syntax

sybernatus committed:
doc: reload pods by using random annotation

Update documentation to update pods using same DAG tag using random airflow annotation instead of manual deleting pods

uranusjr commented on docs/helm-chart/manage-dags-files.rst:
```diff
@@ -71,7 +71,7 @@ Finally, update the Airflow pods with that image:
-Warning :warning:
+.. warning::
 
 *Using constant tag should be used only for testing/development purpose. It is a bad practice to use the same tag as you'll lose the history of your code.*
```
This line needs to be indented so it goes into the warning block. Also please remove the italic syntax (the `*` surrounding the paragraph).

sybernatus committed:
doc: Indent warning message 

indent warning message so it goes in the warning block
remove the italic style

sybernatus committed:
Merge branch 'main' into patch-1

eladkal commented on docs/helm-chart/manage-dags-files.rst:
```diff
@@ -71,12 +71,19 @@ Finally, update the Airflow pods with that image:
-      --set images.airflow.pullPolicy=Always
+      --set images.airflow.pullPolicy=Always \
+      --set airflowPodAnnotations.random=r$(uuidgen)
+      
```
```suggestion
```
Remove whitespace to fix static tests

eladkal committed:
Update docs/helm-chart/manage-dags-files.rst

eladkal commented on docs/helm-chart/manage-dags-files.rst:
```diff
@@ -71,12 +71,18 @@ Finally, update the Airflow pods with that image:
-      --set images.airflow.pullPolicy=Always
+      --set images.airflow.pullPolicy=Always \
+      --set airflowPodAnnotations.random=r$(uuidgen)
+The randomly generated pod annotation will ensure that pods are refreshed on helm upgrade.
```
```suggestion
      --set airflowPodAnnotations.random=r$(uuidgen)

The randomly generated pod annotation will ensure that pods are refreshed on helm upgrade.
```

missing a blank line after markup

eladkal committed:
Update docs/helm-chart/manage-dags-files.rst
